src/Main.elm

```diff
@@ -25,40 +143,939 @@ init =
+                                        "Details Tab: Invalid email"
+                                in
+                                case Util.parseEmail (inputFields |> Dict.get "email" |> Maybe.withDefault "") of
+                                    Ok _ ->
+                                        Set.remove errorMsg Set.empty
+
+                                    Err _ ->
+                                        Set.insert errorMsg Set.empty
+
+                            phoneErrorMsg =
+                                "Details Tab: Valid phone should contain '+' sign followed by 11 min digits"
+
+                            phoneError =
+                                case Util.parsePhoneNumber (inputFields |> Dict.get "phone" |> Maybe.withDefault "") of
+                                    Ok _ ->
+                                        Set.remove phoneErrorMsg Set.empty
+
+                                    Err _ ->
+                                        Set.insert phoneErrorMsg Set.empty
+
+                            madatoryFieldsErrors =
+                                case prefMethod of
+                                    Email ->
+                                        let
+                                            errorMsg =
+                                                "Details Tab: Email is mandatory field"
+                                        in
+                                        if String.isEmpty emailValue then
+                                            Set.insert errorMsg Set.empty
+
+                                        else
+                                            Set.remove errorMsg Set.empty
+
+                                    Post ->
+                                        let
+                                            addressValue =
+                                                inputFields |> Dict.get "address" |> Maybe.withDefault ""
+
+                                            addressErrorMsg =
+                                                "Details Tab: Address is mandatory field"
+                                        in
+                                        if String.isEmpty addressValue then
+                                            Set.insert addressErrorMsg Set.empty
+
+                                        else
+                                            Set.remove addressErrorMsg Set.empty
+
+                                    Phone ->
+                                        let
+                                            errorMsg =
+                                                "Details Tab: Phone is mandatory field"
+                                        in
+                                        if String.isEmpty phoneValue then
+                                            Set.insert errorMsg Set.empty
+
+                                        else
+                                            Set.remove errorMsg Set.empty
```

Comment:
Q: this seems to be repeating quite a lot - what would be a good reason to keep those insert/remove inlined and not refactored/abstracted away

Reply:
Can't say, didn't know better, maybe just inexperienced way of writing